src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -259,7 +264,15 @@ def _normalize_properties(self, user, subscriptions, is_service_principal, cert_
                 _IS_DEFAULT_SUBSCRIPTION: False,
                 _TENANT_ID: s.tenant_id,
                 _ENVIRONMENT_NAME: self.cli_ctx.cloud.name
-            })
+            }
+            # for Subscriptions - List REST API 2019-06-01's subscription account
+            if subscription_dict[_SUBSCRIPTION_NAME] != _TENANT_LEVEL_ACCOUNT_NAME:
+                if hasattr(s, 'subscription_tenant_id'):
+                    subscription_dict[_SUBSCRIPTION_TENANT_ID] = s.subscription_tenant_id
+                if hasattr(s, 'managed_by_tenants'):
```

Comment:
Can we move code like subscription_dict[_SUBSCRIPTION_TENANT_ID] = s.subscription_tenant_id to the initialization code of subscription_dict = {...} to make code more clean?

Reply:
`managed_by_tenants` is only present since API version `2019-06-01`. It is not possible to do conditional initialization according to python syntax.

Reply:
`managed_by_tenants` is only present since API version `2019-06-01`. It is not possible to do conditional initialization according to python syntax. https://stackoverflow.com/a/14263905